server/app/middlewares/authentication.ts

```diff
@@ -20,7 +20,11 @@ export const authenticateJWT = (req: Request, res: Response, next: NextFunction)
       return res.sendStatus(403);
     }
 
-    if (!decoded.username || !decoded.email || !decoded.id) res.sendStatus(403);
+    if (decoded.role === 'General' && NON_GENERAL_PATHS.includes(req.path)) {
+      res.sendStatus(403);
+    }
+
+    if (!decoded.username || !decoded.email || !decoded.id || !decoded.role) res.sendStatus(403);
```

Comment:
I think we can apply DeMorgan's law here to make it look simpler

```suggestion
    if !(decoded.username && decoded.email & decoded.id && decoded.role) res.sendStatus(403);
```

Reply:
But it's up to you really

Reply:
good idea, fixed